### ISSUE TITLE
Boookmarks corrupted by syncing with new PC

### ISSUE THREAD

**Issue by pauldurham** [xbrowsersync/app#74]
### Steps to reproduce
1. Have been running xBrowserSync on PC #1 (Win10, Chrome) for long time, and it syncs correctly.
2. Deleted ALL bookmarks & bookmark bar on PC #2 (Win10, Chrome).
3. Enabled sync with same xBrowserSync ID on PC #2.

### Expected behaviour
I expected the synced bookmarks on PC #1 to remain the same.

### Actual behaviour
The bookmarks on PC #1 have been messed up.  For example some folder names have been duplicated, some bookmarked sites now appear in folders of the wrong name, some folders & sites I bookmarked on PC #1 prior to enabling sync on PC #2 have disappeared.

### Client app information
**App Platform:** (Chrome 64-bit on Windows 10 x64) 

**App Platform Version:** 
Chrome Version 71.0.3578.98 (Official Build) (64-bit), Windows 10 x64 1703.

#### Other info
```
Enter any other information you think would be helpful here.
```

**Comment by pauldurham** [comment]
Correction:
I created an xBrowserSync backup on PC #1 before I enabled syncing with PC #2.  I deleted all bookmarks in PC #1 & restored the backup to PC #1 and the bookmarks are still corrupted or missing.  I suspect therefore that the corruption may have existed before I synced with PC #2.

On PC #1 I have now added the missing folder & bookmarks, renamed some bookmarks, moved some folders and/or bookmarks, and done a general cleanup.  I now see that xBrowswerSync is "stuck" with the message "xBrowserSync - syncing...".  I don't know if I can safely restrt my browser or not, or what I need to do to fix the "syncing" loop.

**Comment by nero120** [comment]
Hi @pauldurham, sorry to hear you are having issues.

I recommend you try the following:

1. Create a new xBrowserSync backup with the cleaned up bookmarks from PC #1
2. On PC #1, [create a new profile](https://www.pcworld.com/article/2089364/how-to-create-and-manage-multiple-user-profiles-in-chrome.html)
3. On the new profile, install the xBrowserSync extension (v1.4.0)
4. Create a new sync (don't create any bookmarks yet)
5. Restore the xBrowserSync backup you created earlier

Hopefully the restore succeeds and you can now see all of your bookmarks as per your original profile. At this point, do the same on PC #2 but at step 4 sync to the new sync ID you created on PC #1. Hopefully you will not get the syncing look issue.

If that fixes it, then you should be able to uninstall then reinstall xBrowserSync from both your original profiles on PC #1 and #2, delete any existing bookmarks and then take it from step 4.

Unfortunately I can't say what caused the original corruption without a detailed account of exactly what was being done when the corruption occurred. If you have any insight as to what was going on at the time around the corruption happened it would be extremely helpful.

**Comment by nero120** [comment]
@pauldurham I've made some major updates to the syncing logic for the upcoming release which I hope will resolve all of your syncing issues. If you can spare a few moments, please give the [v1.5.0 beta release](https://github.com/xbrowsersync/app/releases/tag/v1.5.0beta) a try. It would be amazing if you can confirm whether or not it resolves your issues. 🙏

**Comment by jwhaleyca** [comment]
I have the same issue described above on my Macs & Windows 10 PCs. But it's more complicated as the Chrome bookmarks on a half dozen  Android devices, plus a couple of emulators, are corrupted in the xbrowsersync Android app too. The Android app has multiple entries of the same bookmarks, which is how I 1st noticed the problem. If I was entering a bookmark already entered, the Android app would let me know that and wouldn't let the duplicate bookmark be entered. So I was surprised to see duplicates when searching for a bookmark.

I love this app and it has worked beautifully on all my devices.  I can't think of anything in my environment or approach that has changed to result in this corruption.

Do you have further insights as to how I fix this on my large family of PC's and mobile devices? I will use the beta as suggested on the PC's with Chronium. Do you suggest that I use Brave Browser on the Android devices as I clean this up? 

A possible further wrinkle is I use the Samsung Internet extension as it puts my Chrome bookmarks into a folder in the Samsung Internet browser and its the only efficient:way to clean up Samsung Internet browser bookmarks. I'm trying to limit bookmarks to Chrome only but it's a fairly sizeable project that I don't have much time to focus on. Using the Samsung Internet extension to-date hasn't affected the operation of xbrowsersync.  I should also note that I stopped syncing Chrome bookmarks in my Google account when I started using xbrowsersync.

In any event, if you could expand your beta suggestion a little bit to offer a suggested order to my clean-up that considers the xbrowsersync Android app in the process that would be very much appreciated. Thank you!

**Comment by nero120** [comment]
@pauldurham @jwhaleyca would you mind trying the latest [v1.5.0beta11](https://github.com/xbrowsersync/app/releases/tag/v1.5.0beta)?

Please be aware however, due to a [bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1556427) in Firefox's bookmarks API, changing multiple bookmarks simultaneously will likely result in an error in xBrowserSync. The issues has been fixed in the latest [Firefox Nightly (v69.0a1)](https://www.mozilla.org/en-US/firefox/nightly/all/) so please download and test with Nightly so that you don't hit that known issue.

**Comment by nero120** [comment]
@pauldurham @jwhaleyca I've just released [v1.5.0](https://github.com/xbrowsersync/app/releases/tag/v1.5.0) which resolves these syncing issues. Please upgrade and if you're still experiencing the same issues after upgrading comment here.
